pkg/quota/controller/clusterquotareconciliation/reconciliation_controller.go

```diff
@@ -135,18 +148,52 @@ func (c *ClusterQuotaReconcilationController) Run(workers int, stopCh <-chan str
 	c.queue.ShutDown()
 }
 
-func (c *ClusterQuotaReconcilationController) waitForSyncedStores(ready chan<- struct{}, stopCh <-chan struct{}) {
-	defer utilruntime.HandleCrash()
+// Sync periodically resyncs the controller when new resources are observed from discovery.
+func (c *ClusterQuotaReconcilationController) Sync(discoveryFunc resourcequota.NamespacedResourcesFunc, period time.Duration, stopCh <-chan struct{}) {
+	// Something has changed, so track the new state and perform a sync.
+	oldResources := make(map[schema.GroupVersionResource]struct{})
+	wait.Until(func() {
+		// Get the current resource list from discovery.
+		newResources, err := resourcequota.GetQuotableResources(discoveryFunc)
+		if err != nil {
+			utilruntime.HandleError(err)
+			return
+		}
+
+		// Decide whether discovery has reported a change.
+		if reflect.DeepEqual(oldResources, newResources) {
+			glog.V(4).Infof("no resource updates from discovery, skipping resource quota sync")
+			return
+		}
+
+		// Something has changed, so track the new state and perform a sync.
+		glog.V(2).Infof("syncing resource quota controller with updated resources from discovery: %v", newResources)
```

Comment:
can we bump this to V(4)? probably will be too noisy

Reply:
> can we bump this to V(4)? probably will be too noisy

it's from upstream